service/qos/service_level_controller.cc

```diff
@@ -470,4 +473,15 @@ future<> service_level_controller::unregister_subscriber(qos_configuration_chang
     return _subscribers.remove(subscriber);
 }
 
+future<> service_level_controller::maybe_execute_on_shard0(std::function<future<>(service_level_controller&)> f) {
+    if (dynamic_pointer_cast<raft_service_level_distributed_data_accessor>(_sl_data_accessor)) {
```

Comment:
if (_sl_data_accessor->wants_shard_0) is a little nicer.

Or even do it unconditionally, the non-raft code is transitional.

Reply:
Enabled automatic obtaining of group0 guards and removed `maybe_execute_on_shard0` function